flink-yarn/src/main/java/org/apache/flink/yarn/entrypoint/YarnEntrypointUtils.java

```diff
@@ -51,8 +56,24 @@
  */
 public class YarnEntrypointUtils {
 
-	public static Configuration loadConfiguration(String workingDirectory, Map<String, String> env) {
-		Configuration configuration = GlobalConfiguration.loadConfiguration(workingDirectory);
+	private static final Logger LOG = LoggerFactory.getLogger(YarnEntrypointUtils.class);
+
+	public static Configuration loadConfiguration(String workingDirectory, String[] args, Map<String, String> env) {
+		Configuration dynamicParameters = null;
+		final CommandLineParser<Configuration> commandLineParser = new CommandLineParser<>(
+				new DynamicParametersConfigurationParserFactory());
+
+		try {
+			dynamicParameters = commandLineParser.parse(args);
+		} catch (FlinkParseException e) {
+			LOG.error("Could not parse command line arguments {}.", args, e);
+			commandLineParser.printHelp(StandaloneSessionClusterEntrypoint.class.getSimpleName());
+			System.exit(1);
+		}
```

Comment:
This snipped looks quite similar to what you've put into the `KubernetesEntrypointUtils`. I'd suggest to factor it out and to reuse it. Maybe even pass `dynamicParameters` into this method instead of `args`.

Reply:
You have a good point. I looked into the code again and realized that it's not ideal (the current version and your proposal moving everything in a single utility method). I don't consider calling `System.exit(..)` within a utility method good coding. Such a call should be located in the main method. 

Moving the left-over code into a utility method would have meant splitting up the error handling (since `CommandLineParser.printHelp(..)` would have still been called within the utility method).

I didn't consider splitting up the error handling being a good approach either. Hence, I moved the dynamic parameter parsing into each main method in [7f722c9](https://github.com/apache/flink/pull/13747/commits/7f722c99b83591793147b2dda952a3acadf5f2b7). What's your take on this?

FYI: The commit above also fixed a bug that used the wrong class in `CommandLineParser.printHelp(..)`.